Discord login

bates64 opened the pull request.

Commit:
format all the things (#2)

L5050 committed:
files for mod list

Commit:
Merge branch 'main' into dev

Commit:
simpler mods page

L5050 committed:
added login button

L5050 committed:
added login page with discord auth

L5050 committed:
login with discord

L5050 committed:
fix file tree

L5050 committed:
fixed a few issues, redirect still doesn't work

L5050 committed:
reverted env change

L5050 committed:
cleaning up

L5050 committed:
full login and logout functionality

Commit:
format

Commit:
hide mods link

L5050 committed:
added component, imported to main file

L5050 committed:
format

L5050 committed:
fixed redirect

L5050 committed:
imported login to mod and games directory

bates64 commented on src/hooks.server.ts:
```diff
@@ -0,0 +1,27 @@
+    SvelteKitAuth({
+        providers: [
+            Discord({
+                clientId: "1113277158543073381",
```
Should be from env

bates64 commented on src/lib/components/Signin.svelte:
```diff
@@ -0,0 +1,61 @@
+
+    let showSignOutButton = false;
+
+    function signIn() {
+        window.location.href = "/auth/signin";
+    }
+
+    function signOut() {
+        window.location.href = "/auth/signout";
+    }
```
There may be a "proper" way to do redirects with sveltekit, can you check?

bates64 replied:
goto from $app/navigation

bates64 replied:
Unless this breaks auth.js

L5050 replied:
theres the goto function but it sometimes doesn't redirect properly

bates64 commented on src/lib/components/Signin.svelte:
```diff
@@ -0,0 +1,61 @@
+    {#if $page.data.session.user?.image}
+        <div
+            class="user-icon"
+            style="background-image: url('{$page.data.session.user.image}');"
```
I think `style:background-image=` is better

bates64 commented on src/lib/components/Signin.svelte:
```diff
@@ -0,0 +1,61 @@
+        <div
+            class="user-icon"
+            style="background-image: url('{$page.data.session.user.image}');"
+            on:click={() => (showSignOutButton = true)}
```
```suggestion
            on:click={() => {showSignOutButton = true}}
```

bates64 commented on src/lib/components/Signin.svelte:
```diff
@@ -0,0 +1,61 @@
+{/if}
+
+<style>
+    .user-icon {
+        position: fixed;
+        top: 20px;
+        right: 20px;
+        width: 50px;
+        height: 50px;
+        border-radius: 50%;
+        background-size: cover;
+        background-position: center;
+        cursor: pointer;
+        border: 2px solid black;
+    }
+    .signin-button {
+        position: fixed;
+        top: 10px;
+        right: 10px;
+        cursor: pointer;
+        background-color: #444;
+        color: #fff;
+        padding: 10px;
+        border-radius: 5px;
+    }
+    .sign-out-button {
+        position: fixed;
+        top: 80px;
+        right: 20px;
+    }
```
Look into doing this with https://tailwindcss.com/ instead please

bates64 commented on src/routes/+page.svelte:
```diff
@@ -3,7 +3,7 @@
     import LandingHeader from "./LandingHeader.svelte";
     import LinkButton from "./LinkButton.svelte";
-
+    import Signin from "$lib/components/Signin.svelte";
```
Can you name the component `SignIn` (i.e. capitalise "in")

L5050 replied:
Sure yeah

bates64 commented on src/routes/mods/+page.svelte:
```diff
@@ -0,0 +1,92 @@
+	import { page } from "$app/stores";
+	let showSignInButton = false;
+	import Signin from "$lib/components/Signin.svelte";
+	const mods = [
+	{
+	name: "Paper Mario: Master Quest",
+	description:
+	'Considered "The Dark Souls of Paper Mario", Master Quest is incredibly difficult. It was designed with elite challenge runners in mind, and unless you have extensive knowledge of Paper Mario, you may struggle with much of the content.',
+	link: "https://github.com/Brotenko/PMMasterQuest",
+	},
+	{
+	name: "Paper Mario Mod 2",
+	description: "This is a description of Mod 2.",
+	link: "#",
+	},
+	];
+
+	function signIn() {
+	window.location.href = "/auth/signin";
+	}
+	function signOut() {
+	window.location.href = "/auth/signout";
+	}
```
Needs formatting, do `pnpm run format` and it should hopefully fix this
if not thats a bug 💀

bates64 commented on src/routes/mods/+page.svelte:
```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+	import Breadcrumbs from "$lib/components/Breadcrumbs.svelte";
+	import { page } from "$app/stores";
+	let showSignInButton = false;
+	import Signin from "$lib/components/Signin.svelte";
```
put imports at the top like
```suggestion
	import Signin from "$lib/components/Signin.svelte";
    let showSignInButton = false;
```

bates64 commented on src/routes/mods/+page.svelte:
```diff
@@ -0,0 +1,92 @@
+    .mod-description {
+        color: #666;
+    }
+</style>
```
again tailwind where possible

Commit:
add Button component

L5050 committed:
fixed env bug

Commit:
Merge branch 'main' into dev

Commit:
Merge branch 'dev' of github.com:star-haven/starhaven.dev into dev

L5050 committed:
formatting

L5050 committed:
made Sign out a button

Commit:
fix file capitalisation

Commit:
Merge branch 'dev' of github.com:star-haven/starhaven.dev into dev

Commit:
add sign in test

Doesn't work for now due to auth.js issue, also will need a test discord account but we can fix it later

Commit:
update logotype

Commit:
add favicon

Doesn't appear to work though :/

Commit:
use tailwind for signin nav

Commit:
add editorconfig

Commit:
move logotype

Commit:
simplify auth hook

Commit:
remove unused component

Commit:
improve session nav

L5050 committed:
format and profile page

L5050 committed:
format + tailwind for profile page

L5050 committed:
merged layout and page.svelte

L5050 reviewed (approved):
adds new login features and a profile page
